instrumentation/net_http/lib/opentelemetry/instrumentation/net/http/patches/instrumentation.rb

```diff
@@ -14,10 +14,16 @@ module Instrumentation
             HTTP_METHODS_TO_SPAN_NAMES = Hash.new { |h, k| h[k] = "HTTP #{k}" }
             USE_SSL_TO_SCHEME = { false => 'http', true => 'https' }.freeze
 
-            def request(req, body = nil, &block)
+            def request(req, body = nil, &block) # rubocop:disable Metrics/MethodLength
               # Do not trace recursive call for starting the connection
               return super(req, body, &block) unless started?
 
+              if untraced?
+                OpenTelemetry::Common::Utilities.untraced do
+                  return super(req, body, &block)
+                end
+              end
```

Comment:
Do you need the `untraced` block here? Its only purpose is to avoid creating spans along a codepath where you may not have full control of the instrumentation. In this case, there should be no "deeper" instrumentation than the `kind=client` span we're creating below. I think it should be enough to:
```suggestion
              return super(req, body, &block) if untraced?
```